docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md

```diff
@@ -0,0 +1,203 @@
+---
+approvers:
+- pipejakob
+- luxas
+- roberthbailey
+- jbeda
+title: Upgrading kubeadm clusters
+---
+
+{% capture overview %}
+
+This guide is for upgrading `kubeadm` clusters from version 1.7.x to 1.8.x.
+See also [upgrading kubeadm clusters from 1.6 to 1.7](/docs/tasks/administer-cluster/kubeadm-upgrade-1-7/) if you're on a 1.6 cluster currently.
+
+{% endcapture %}
+
+{% capture prerequisites %}
+
+Before proceeding:
+
+- You need to have a functional `kubeadm` Kubernetes cluster running version 1.7.0 or higher in order to use the process described here.
+- Make sure you read the [release notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG.md#v180-beta1) carefully.
+- As `kubeadm upgrade` does not upgrade etcd make sure to back it up. You can, for example, use `etcdctl backup` to take care of this.
+- Note that `kubeadm upgrade` will not touch any of your workloads, only Kubernetes-internal components. As a best-practice you should back up what's important to you. For example, any app-level state, such as a database an app might depend on (like MySQL or MongoDB) must be backed up beforehand.
+
+Also, note that only one minor version upgrade is supported. That is, you can only upgrade from, say 1.7 to 1.8, not from 1.7 to 1.9.
+
+{% endcapture %}
+
+{% capture steps %}
+
+## Upgrading your control plane
+
+You have to carry out the following steps by executing these commands on your master node:
+
+1. Install the most recent version of `kubeadm` using `curl` like so:
+
+```shell
+$ export VERSION=v1.8.0 # or any given released Kubernetes version
+$ export ARCH=amd64 # or: arm, arm64, ppc64le, s390x
+$ curl -sSL https://dl.k8s.io/release/${VERSION}/bin/linux/${ARCH}/kubeadm > /usr/bin/kubeadm
+```
+
+1. On the master node, run the following:
+
+```shell
+$ kubeadm upgrade plan
```

Comment:
First of all, you have to remember what CLI args you passed to `kubeadm init` the first time.
If you used flags, do this `kubeadm config upload from-flags [flags]`, flags can be empty
If you used a config file, do this `kubeadm config upload from-file --config [config]`, config mandatory

This is to preserve the config for future upgrades. Note that this has only to be done on the **first** time you use `kubeadm upgrade`